parsec/backend/invite.py

```diff
@@ -191,18 +193,19 @@ def generate_invite_email(
 ) -> Message:
     mail_text = MAIL_TEXT[email_config.language]
     parsec_url = "https://parsec.cloud/get-parsec"
-    invite_link = str(
-        BackendInvitationAddr(
-            organization_id=organization_id,
-            invitation_type=InvitationType.USER,
-            token=invitation.token,
-            hostname=backend_addr.hostname,
-            port=backend_addr.port,
-            use_ssl=backend_addr.use_ssl,
-        )
+    backend_invitation_addr = BackendInvitationAddr.build(
+        backend_addr=backend_addr,
+        organization_id=organization_id,
+        invitation_type=InvitationType.USER,
```

Comment:
`generate_invite_email` is used for both user and device invitation (the signature of the invitation parameter is wrong, it should be `invitation: Invitation`)

Hence we should deduce the type of the invitation from the `invitation` parameter:
```python
invitation_type=InvitationType.USER if isinstance(invitation, UserInvitation) else InvitationType.DEVICE
```

However I think it would be better to add a `UserInvitation.type` and `DeviceInvitation.type` constant attribute (the if/else sentence is cumbersome and error prone if another kind of invitation get introduced in the future):
```python
class UserInvitation:
    TYPE = InvitationType.USER  # Note this is a class attribute (not to be confounded with the instance attributes declared in a similar fashion with class decorated with `@attr.s`)
```

Reply:
Done 👍